python/cudf/dataframe/dataframe.py

```diff
@@ -722,7 +722,7 @@ def index(self):
     @index.setter
     def index(self, _index):
         if isinstance(_index, cudf.dataframe.multiindex.MultiIndex):
-            if len(_index) != len(self):
+            if len(_index) != len(self[self.columns[0]]):
```

Comment:
@rjzamora Could you update your branch with the latest changes from cudf. New changes recently got merged.